codespell_lib/_codespell.py

```diff
@@ -983,6 +984,20 @@ def parse_file(
     return bad_count
 
 
+def flatten_comma_separated_arguments(
+    arguments: List[str],
+) -> List[str]:
+    """
+    >>> flatten_comma_separated_arguments(["a,b,c", "d", "e"])
+    ['a', 'b', 'c', 'd', 'e']
+    >>> flatten_comma_separated_arguments([])
+    []
+    >>> flatten_comma_separated_arguments(None)
```

Comment:
This one shouldn't be allowed based on the type hints (you'd need an `Optional`), and it's not used in code. Let's remove it

Reply:
I understand, but unfortunately mypy is unable to enforce all hints, and I have always read that Pythonic functions should be lenient (probably read that before mypy).

Reply:
Sure but we don't need to be lenient here. I would only add leniency when it's needed. To put it differently, the `or []` branch isn't useful now, so why add it now? It seems like over-planning for a future, i.e., YAGNI